compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -1502,6 +1502,9 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                                 pluralize!(count),
                             ),
                         );
+                        if sp.is_desugaring(DesugaringKind::Async) {
+                            err.note("while checking the return type of the `async fn`");
```

Comment:
I think you'll have to keep a local that checks if you have noted this in this loop already
In the rebase on my pr, this message shows up twice in 1 diagnostic

Reply:
Will do!

Reply:
I will experiment with this (likely tomorrow), but is there any chance you could point me to a place in the code base where it does a similar local check in the loop?

Reply:
Experimented and I think I've got what you suggested working. Would you mind trying it out when you have a moment?